rclcpp/include/rclcpp/publisher_options.hpp

```diff
@@ -80,10 +85,7 @@ struct PublisherOptionsWithAllocator : public PublisherOptionsBase
   to_rcl_publisher_options(const rclcpp::QoS & qos) const
   {
     rcl_publisher_options_t result = rcl_publisher_get_default_options();
-    using AllocatorTraits = std::allocator_traits<Allocator>;
-    using MessageAllocatorT = typename AllocatorTraits::template rebind_alloc<MessageT>;
-    auto message_alloc = std::make_shared<MessageAllocatorT>(*this->get_allocator().get());
-    result.allocator = rclcpp::allocator::get_rcl_allocator<MessageT>(*message_alloc);
+    result.allocator = rclcpp::allocator::get_rcl_allocator<void>(*this->get_allocator());
```

Comment:
Hmm, I think this was a mistake. From `cppreference`:

> The explicit specialization for void lacks the member typedefs reference, const_reference, size_type and difference_type. This specialization declares no member functions.

It appears `void` allocators are not supposed to be used for anything but type-erasure. If everything complies with that i.e. no code attempts to call `allocate` on a `void` allocator then you can afford not making the explicit specialization like the test allocator, the allocator tutorial and now the tlsf allocator do...

Reply:
can you fix that one?

Reply:
#1657 is up.